examples/dreambooth/train_dreambooth_lora_hidream.py

```diff
@@ -253,8 +248,7 @@ def log_validation(
             )
 
     del pipeline
-    if torch.cuda.is_available():
-        torch.cuda.empty_cache()
+    free_memory()
```

Comment:
If we're doing pre-encoding of prompts, then it is important to first move the model components of the `pipeline` (the `pipeline` object for doing prompt encoding) to CPU and then perform the rest of the stuff for memory savings. 

Reference:
https://github.com/huggingface/diffusers/issues/11376#issuecomment-2820351451